### PR TITLE
Hackathon pre BIG merge

### PR DESCRIPTION
**Pull request by schuyler1d**
This is an even bigger 'master' merge which includes #407 (and thus #360  #371 and #368 and the changes in #369 or #370)

It also merges 
* #406 (jmcarp:fix-empty-contact-count)
* #404 (jmcarp:codeshift-prop-types)
* #402 (jmcarp:update-types)
* #398 (jmcarp:eslint)
* #394 (jmcarp:upgrade-webpack)
* #390 (jmcarp:babel-preset-env)
* #388 (jmcarp:update-shrinkwrap)

Mostly this is just resolving conflicts, but 15cdeb9b8951471d8d8a149a85e88c6961aeef6c resolves some issues introduced in some of the grapql/babel upgrades which made apollo a little more 'rigorous' in checking schemas (@jmcarp )
Also includes some additional linting fixes.

### CODE REVIEW

File: src/containers/AssignmentTexterContact.jsx [hunk 1]
```diff
@@ -249,7 +255,6 @@ class AssignmentTexterContact extends React.Component {
 
   getAvailableInteractionSteps(questionResponses) {
```
Comment (shakalee14) [inline src/containers/AssignmentTexterContact.jsx:256]: Still getting an error on this screen:
```
10:51:13 AM server.1              |  TypeError: Cannot read property '0' of undefined
10:51:13 AM server.1              |      at getTopMostParent (/Users/shaka/moveon/Spoke/src/lib/interaction-step-helpers.js:83:10)
10:51:13 AM server.1              |      at _callee7$ (/Users/shaka/moveon/Spoke/src/server/api/campaign-contact.js:168:14)
10:51:13 AM server.1              |      at tryCatch (/Users/shaka/moveon/Spoke/node_modules/regenerator-runtime/runtime.js:65:40)
10:51:13 AM server.1              |      at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/shaka/moveon/Spoke/node_modules/regenerator-runtime/runtime.js:303:22)
10:51:13 AM server.1              |      at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/shaka/moveon/Spoke/node_modules/regenerator-runtim
10:51:13 AM server.1              |  >  e/runtime.js:117:21)
10:51:13 AM server.1              |      at step (/Users/shaka/moveon/Spoke/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
10:51:13 AM server.1              |      at /Users/shaka/moveon/Spoke/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
10:51:13 AM server.1              |      at run (/Users/shaka/moveon/Spoke/node_modules/core-js/modules/es6.promise.js:66:22)
10:51:13 AM server.1              |      at /Users/shaka/moveon/Spoke/node_modules/core-js/modules/es6.promise.js:79:30
10:51:13 AM server.1              |      at flush (/Users/shaka/moveon/Spoke/node_modules/core-js/modules/_microtask.js:18:9)
10:51:13 AM server.1              |      at _combinedTickCallback (internal/process/next_tick.js:67:7)
10:51:13 AM server.1              |      at process._tickDomainCallback (internal/process/next_tick.js:122:9)
```

Comment (shakalee14) [reply src/containers/AssignmentTexterContact.jsx:256]: Which I think has to deal with interactionSteps still coming in as an empty array

**Comment by sreynen** [comment]
I get an error running `npm run prod-build-server` with this PR, which I don't get with master: `Error: Couldn't find preset "env" relative to directory`. I also get an error running `npm run prod-build-client`, also not seen with master: `TypeError: webpack.LoaderOptionsPlugin is not a constructor`

File: src/containers/AssignmentTexterContact.jsx (hunk 1 above)
Comment (schuyler1d) [reply src/containers/AssignmentTexterContact.jsx:256]: @shakalee14 I'll look into this more, but do you still get this from a fresh campaign or one that might have been created with the previous version?

**Comment by schuyler1d** [comment]
@sreynen This PR contains a bunch of changes to package.json, so it's important to remove your current `node_modules` directory and run `npm install` from scratch again.  With that done, I don't get an error.

File: src/containers/AssignmentTexterContact.jsx (hunk 1 above)
Comment (shakalee14) [reply src/containers/AssignmentTexterContact.jsx:256]: this is with a fresh campaign - when attempting to use dynamic assignment

**Comment by schuyler1d** [comment]
This also fixes #417

**Comment by shakalee14** [comment]
Mobile Regression for Replies - as seen on all mobile devices:
This bottom set of buttons is rendered via `AssignmentTexterContact.jsx`L599
screen shot 2017-12-06 at 11 40 28 am

**Comment by shakalee14** [comment]
Assignment Regression for Non-dynamic Texter Assignment Creation:

User story: I'm an admin, create a campaign and assign a texter 3 contacts. I start the campaign and then need to edit that texter assignment to 0 and assign another texter those 3 contacts. My new assignment isn't saving.

This works locally but doesn't work for me on stage.

**Comment by shakalee14** [comment]
Regression: 
Locally - texter reply screen does not load. This error is in the client:
`_timezones.getLocalTime)(...).format is not a function`

Stage - texter reply screen loads but city and state location return 'null null' in the UX

In the client react developer tools, you can see an object like the following for the campaignContact:
```
campaignContact:
{…}
assignmentId: "126"
cell: "+xxxxxxxxx"
currentInteractionStepScript:
"hey {firstName} - this is {texterFirstName}!"
customFields: "{\"home_state\":\"il\",\"city\":\"chicago\",\"event_id\":\"xxxx\",\"event_page\":\"xxxx\"}"
firstName: "name"
id: "34462"
interactionSteps: Array[1]
lastName: "last name"
location:
{…}
city: null
state: null
timezone: {…}
etc ... ```

**Comment by schuyler1d** [comment]
https://github.com/MoveOnOrg/Spoke/pull/411#issuecomment-349747187 was resolved with 
https://github.com/MoveOnOrg/Spoke/pull/411/commits/592aad410dd23391cd4cfb9d02a332ec2a02317a
It surfaced because we don't have the zip_code table populated on demo OR production (we should fix that :-)
The fix falls back more gracefully.

**Pull request merged**